internal/gatewayapi/runner/runner.go

```diff
@@ -310,6 +317,70 @@ func (r *Runner) subscribeAndTranslate(sub <-chan watchable.Snapshot[string, *re
 	r.Logger.Info("shutting down")
 }
 
+func (r *Runner) preservePolicyStatusesCrossGatewayClasses(resources *resource.Resources) {
+	for _, policy := range resources.ClientTrafficPolicies {
+		if policyStatus, ok := r.ProviderResources.ClientTrafficPolicyStatuses.Load(utils.NamespacedName(policy)); ok {
+			policy.Status = *policyAncestorsFromOtherGatewayClasses(policyStatus, policy.Namespace, resources)
+		}
+	}
+	for _, policy := range resources.BackendTrafficPolicies {
+		if policyStatus, ok := r.ProviderResources.BackendTrafficPolicyStatuses.Load(utils.NamespacedName(policy)); ok {
+			policy.Status = *policyAncestorsFromOtherGatewayClasses(policyStatus, policy.Namespace, resources)
+		}
+	}
+	for _, policy := range resources.SecurityPolicies {
+		if policyStatus, ok := r.ProviderResources.SecurityPolicyStatuses.Load(utils.NamespacedName(policy)); ok {
+			policy.Status = *policyAncestorsFromOtherGatewayClasses(policyStatus, policy.Namespace, resources)
+		}
+	}
+	for _, policy := range resources.EnvoyExtensionPolicies {
+		if policyStatus, ok := r.ProviderResources.EnvoyExtensionPolicyStatuses.Load(utils.NamespacedName(policy)); ok {
+			policy.Status = *policyAncestorsFromOtherGatewayClasses(policyStatus, policy.Namespace, resources)
+		}
+	}
+	for _, policy := range resources.EnvoyPatchPolicies {
+		if policyStatus, ok := r.ProviderResources.EnvoyPatchPolicyStatuses.Load(utils.NamespacedName(policy)); ok {
+			policy.Status = *policyAncestorsFromOtherGatewayClasses(policyStatus, policy.Namespace, resources)
+		}
+	}
+	for _, policy := range resources.ExtensionServerPolicies {
+		key := message.NamespacedNameAndGVK{
+			NamespacedName:   utils.NamespacedName(&policy),
+			GroupVersionKind: policy.GroupVersionKind(),
+		}
+		if policyStatus, ok := r.ProviderResources.ExtensionPolicyStatuses.Load(key); ok {
+			policy.Object["status"] = policyAncestorsFromOtherGatewayClasses(policyStatus, policy.GetNamespace(), resources)
+		}
+	}
+	for _, policy := range resources.BackendTLSPolicies {
+		if policyStatus, ok := r.ProviderResources.BackendTLSPolicyStatuses.Load(utils.NamespacedName(policy)); ok {
+			policy.Status = *policyAncestorsFromOtherGatewayClasses(policyStatus, policy.Namespace, resources)
+		}
+	}
+}
+
+func policyAncestorsFromOtherGatewayClasses(policyStatus *gwapiv1a2.PolicyStatus, policyNamespace string, resources *resource.Resources) *gwapiv1a2.PolicyStatus {
+	AncestorsFromOtherGatewayClasses := make([]gwapiv1a2.PolicyAncestorStatus, 0)
+	for _, ancestor := range policyStatus.Ancestors {
+		if ancestor.AncestorRef.Group == nil ||
+			*ancestor.AncestorRef.Group == "gateway.networking.k8s.io" {
```

Comment:
This assumes all xPolicy types use `Gateway` as the status ancestor, which is recommended by the Gateway API spec.
We may need to clean up the xPolicy status in a follow-up PR.